Aptoide/aptoide-client-v8
# [MOB-72-74] Download and Update to room

**What does this PR do?**

This PR aims at migrating update and download tables from Realm to Room. These had to be migrated at the same time because of the fact that both depend on the Splits table.
Note: some small refactors were made as some methods, for example, were Observable without needing to. Also, **due to the fact that we were based on some realm features (like returning null, in case the query is empty)** we had the change some of the behavior.

PS: While testing this, i noticed that sometimes, in Xiaomi devices, we don't get the callback from Android indicating that an app was unninstalled, udpated or installed.

**Database changed?**

   Yes

**Where should the reviewer start?**

- [x] RoomDownloadPersistence.java

**How should this be manually tested?**

Open the app, try to perform both downloads and updates. Please check if performance was affected. Test splits, autoupdate and the normal flows aswell. Take into consideration the PS note added on top, regarding the xiaomi devices.

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-72](https://aptoide.atlassian.net/browse/MOB-72)
[MOB-74](https://aptoide.atlassian.net/browse/MOB-74)


**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
